driver/Driveraux.ml

```diff
@@ -14,12 +14,33 @@
 open Printf
 open Clflags
 
+(* Is this a gnu based tool chain *)
+let gnu_system = Configuration.system <> "diab"
+
 (* Invocation of external tools *)
 
 let rec waitpid_no_intr pid =
   try Unix.waitpid [] pid
   with Unix.Unix_error (Unix.EINTR, _, _) -> waitpid_no_intr pid
 
+let responsefile args resp_arg =
+  try
+    if Sys.win32 && (String.length (String.concat "" args) > 7000) then
```

Comment:
Could do `List.fold_left (fun len arg -> len + String.length arg + 1) 0 args > 7000` here, to avoid constructing the concatenated string.  Also, one space should be counted between every argument, I think.  Finally, I vaguely remember that Cygwin has its own limitations on the command-line length, not as strict as Win32's, but not as big as Linux or MacOS.